dallinger/frontend/static/scripts/dallinger2.js

```diff
@@ -307,34 +392,119 @@ var dallinger = (function () {
     return deferred;
   };
 
+  /**
+   * Creates a new experiment node for the current partcipant.
+   *
+   * @example
+   * var response = dallinger.createAgent();
+   * // Wait for response
+   * response.done(function (data) {... handle data.node ...});
+   *
+   * @returns {jQuery.Deferred} See :ref:`deferreds-label`
+   */
   dlgr.createAgent = function () {
     return dlgr.post('/node/' + dallinger.identity.participantId);
   };
 
+  /**
+   * Creates a new info object in the experiment database.
+   *
+   * @example
+   * var response = dallinger.createInfo(1, {details: {a: 1}});
+   * // Wait for response
+   * response.done(function (data) {... handle data.info ...});
+   *
+   * @param {number} nodeId - The id of the participant's experiment node
+   * @param {Object} data - Experimental data (see :class:`~dallinger.models.Info`)
+   * @returns {jQuery.Deferred} See :ref:`deferreds-label`
+   */
   dlgr.createInfo = function (nodeId, data) {
     return dlgr.post('/info/' + nodeId, data);
   };
 
+  /**
+   * Returns a public property value for the experiment.
+   *
+   * @example
+   * var response = dallinger.getExperimentProperty("propname");
+   * // Wait for response
+   * response.done(function (data) {... handle e.g. data.propname ...});
+   *
+   * @param {string} prop - The experiment property to lookup
+   * @returns {jQuery.Deferred} See :ref:`deferreds-label`
+   */
   dlgr.getExperimentProperty = function (prop) {
     return dlgr.get('/experiment/' + prop);
   };
 
+  /**
+   * Get a specific info object.
+   *
+   * @example
+   * var response = dallinger.getInfo(1, 1);
+   * // Wait for response
+   * response.done(function (data) {... handle data.info ...});
+   *
+   * @param {number} nodeId - The id of an experiment node
+   * @param {number} infoId - The id of the Info object to be retrieved
+   * @returns {jQuery.Deferred} See :ref:`deferreds-label`
+   */
   dlgr.getInfo = function (nodeId, infoId) {
     return dlgr.get('/info/' + nodeId + '/' + infoId);
   };
 
+  /**
+   * Get all infos for the specified node.
+   *
+   * @example
+   * var response = dallinger.getInfos(1, 1);
+   * // Wait for response
+   * response.done(function (data) {... handle data.infos ...});
```

Comment:
A big job, and maybe brittle: to document the format of return values 😬 Seems like a separate story, if it seemed valuable enough.

Reply:
This documentation might be better done on the server endpoints/routes.

Reply:
I think that really belongs in the API route documentation, since it's not specific to the JS API.

Reply:
I will add a little clarification on the value passed to the callback though.